src/composables/fungibleTokens.ts

```diff
@@ -0,0 +1,491 @@
+        { decodedResult: token0 },
+        { decodedResult: token1 },
+        { decodedResult: reserves },
+        { decodedResult: totalSupply },
+      ] = await Promise.all([
+        tokenContract.methods.balances(),
+        tokenContract.methods.balance(activeAddress),
+        tokenContract.methods.token0(),
+        tokenContract.methods.token1(),
+        tokenContract.methods.get_reserves(),
+        tokenContract.methods.total_supply(),
+      ]);
+
+      return {
+        token0: {
+          ...availableTokens.value?.[token0],
+          amount: calculateSupplyAmount(
+            balance, totalSupply, reserves.reserve0,
+          ) as number,
+          reserve: reserves.reserve0,
+        },
+        token1: {
+          ...availableTokens.value?.[token1],
+          amount: calculateSupplyAmount(
+            balance, totalSupply, reserves.reserve1,
+          ) as number,
+          reserve: reserves.reserve1,
+        },
+        totalSupply,
+        balance,
+        balances,
+      };
+    } catch (error) {
+      handleUnknownError(error);
+      return null;
+    }
+  }
+
+  async function transferTokens({
+    contractId,
+    recipient,
+    amount,
+    options,
+  }: ITransferTokensOptions) {
+    const sdk = await getSdk();
+    const tokenContract = await sdk.getContractInstance({
+      source: FUNGIBLE_TOKEN_CONTRACT,
+      contractAddress: contractId,
+    });
+    return tokenContract.methods.transfer(recipient, amount.toFixed(), options);
+  }
+
+  async function burnTriggerPoS({
+    contractId,
+    amount,
+    posAddress,
+    invoiceId,
+    options,
+  }: IBurnTriggerPosOptions) {
+    const sdk = await getSdk();
+    const tokenContract = await sdk.getContractInstance({
+      source: ZEIT_TOKEN_INTERFACE,
+      contractAddress: contractId,
+    });
+    return tokenContract.methods.burn_trigger_pos(
+      amount.toFixed(), posAddress, invoiceId, options,
+    );
+  }
+
+  /**
+   * Returns an array of transfer transactions for the specified address,
+   * either from the `transactions`
+   * state or by making a request to the middleware server.
+   * If `recent` is true, the function fetches
+   * the latest transfer transactions for the address,
+   * up to the last transaction in the `transactions`
+   * state. If `recent` is false or `transactions.value[address]` is empty, the function fetches all
+   * transfer transactions for the address. The resulting array of transactions is stored in the
+   * `transactions` state for the address.
+   *
+   * @param {string} options.address - The address for which to fetch transfer transactions.
+   * @param {boolean} options.recent - If true, fetches only the latest transactions.
+   * @returns {Promise<ITransaction[]>} - An array of transactions for the specified address.
+   * @throws {Error} - If an unknown error occurs during the request.
+   */
+  async function getTokensHistory(address: string, recent = false): Promise<ITransaction[]> {
```

Comment:
Little suggestion about naming pattern I would like to have in our app. Functions that are asynchronous (fetches something) should start with `fetch...`. Functions that parses existing data should start with `get...`. So in this case i suggest to use `fetch...`.